npm/webpack-dev-server/src/helpers/vueCliHandler.ts

```diff
@@ -1,25 +1,27 @@
 import debugLib from 'debug'
-import type { Configuration } from 'webpack'
+//import type { Configuration } from 'webpack'
 import type { PresetHandlerResult, WebpackDevServerConfig } from '../devServer'
 import { sourceDefaultWebpackDependencies } from './sourceRelativeWebpackModules'
 
 const debug = debugLib('cypress:webpack-dev-server:vueCliHandler')
 
-export function vueCliHandler (devServerConfig: WebpackDevServerConfig): PresetHandlerResult {
+export async function vueCliHandler (devServerConfig: WebpackDevServerConfig): Promise<PresetHandlerResult> {
   const sourceWebpackModulesResult = sourceDefaultWebpackDependencies(devServerConfig)
 
   try {
-    const config = require.resolve('@vue/cli-service/webpack.config', {
+    const Service = require(require.resolve('@vue/cli-service/lib/Service', {
```

Comment:
Interesting, I have not looked at the Vue CLI code in some time. What about users on older versions of Vue CLI? Will they have a problem?

We have tests around this https://github.com/cypress-io/cypress/tree/develop/system-tests/projects/vuecli4-vue2 Vue CLI 4 and Vue 2. Amazingly they are all passing - but I don't understand how this can be the case, unless `service/lib/Service` has always been a thing?

Reply:
Well, the v3 branch of vue-cli contains [the file](https://github.com/vuejs/vue-cli/blob/v3/packages/%40vue/cli-service/lib/Service.js) and to me it does not seem like much of the called code has changed.

I think #23605 would allow to move the fix to vue-cli code (e.g. return a promise from their `webpack.config.js`), but the main aim of this pull request is to get the default configuration to work with a `vue.config.js` and `"type": "module"` and I do not know if there are other conflicts on the vue-cli side.